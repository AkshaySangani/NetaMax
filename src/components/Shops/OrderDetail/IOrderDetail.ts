export interface IOrderDetail {
  /**
   * The items's name
   * @type {String}
   *
   * */
  name: string;

  /**
   * The items's quantity bought
   * @type {String}
   *
   * */
  qty: number;

  /**
   * The items's price
   * @type {string}
   *
   * */
  price: string;

  /**
   * The items's image
   * @type {String}
   *
   * */
  image: string;

  /**
   * The items's image format
   * @type {String}
   *
   * */
  imageFormat: string;
}
