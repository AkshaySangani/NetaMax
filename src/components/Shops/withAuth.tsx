import React from "react";

import { useSelector } from "react-redux";

import { LayoutShopsContainer } from "containers/Shops/Layout/ShopsLayoutContainer";
import { ShopsLoginContainer } from "containers/Shops/ShopsLogin/ShopsLoginContainer";
import { selectIsAuth } from "dataflows/Shops/Login/LoginSelectors";
import { NextComponentType } from "next";

/**
 * WithAuth HOC
 * @param {NextComponentType} Component component to check
 * @returns {React.FC<T>} React FC
 */
function withAuth<T>(Component: NextComponentType<T>): React.FC<T> {
  /**
   * Check if shop user is authenticated
   * @param {T} props component props
   * @returns {NextComponentType} Component
   */
  const Auth = (props: T) => {
    const isAuth = useSelector(selectIsAuth);

    // If user is not logged in, return shops login component
    if (!isAuth) {
      return <ShopsLoginContainer />;
    }

    // If user is logged in, return original component
    return (
      <LayoutShopsContainer>
        <Component {...props} />
      </LayoutShopsContainer>
    );
  };

  // Copy getInitial props so it will run as well
  if (Component.getInitialProps) {
    Auth.getInitialProps = Component.getInitialProps;
  }

  return Auth;
}

export default withAuth;
