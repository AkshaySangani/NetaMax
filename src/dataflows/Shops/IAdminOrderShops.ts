export interface IAdminOrderShops {
  /**
   * id product
   * @type {number}
   * */
  id: number;
  /**
   * order's orderId
   * @type {string}
   * */
  orderId: string;
  /**
   * product name
   * @type {string}
   * */
  name: string;
  /**
   * product price
   * @type {string}
   * */
  price: string;
  /**
   * qty's product
   * @type {number}
   * */
  qty: number;
  /**
   * image product
   * @type {string | null}
   * */
  image: string | null;
  /**
   * image Format
   * @type {string | null}
   * */
  imageFormat: string | null;
  /**
   * Order Customer Phone
   * @type {String}
   *
   * */
  phone: string;
  /**
   * Order's subtotal
   * @type {Number}
   *
   * */
  subtotal: number;
  /**
   * Order's discount
   * @type {Number}
   *
   * */
  discount: number;
  /**
   * Picked up date
   * @type {string}
   *
   * */
  pickedUpDateAt?: string;
}
