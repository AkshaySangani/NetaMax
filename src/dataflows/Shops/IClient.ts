export interface IClient {
  /**
   * Client's id
   * @type {string}
   */
  id: string;

  /**
   * Client's username
   * @type {string}
   */
  userName: string;

  /**
   * Client's phone number
   * @type {string}
   */
  phone: string;

  /**
   * Client's customer id
   * @type {string}
   */
  customerId: string;

  /**
   * Client's number of orders
   * @type {number}
   */
  numberOfOrders: number;

  /**
   * Client's days ago from last order
   * @type {number}
   */
  daysAgoLastOrder: number;

  /**
   * The Customer's Store URL
   * @type {String}
   *
   * */
  url: string;
}
