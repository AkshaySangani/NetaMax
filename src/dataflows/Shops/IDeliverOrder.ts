export interface IDeliverOrder {
  /**
   * Deliver Order's id
   * @type {string}
   */
  id: string;
}
