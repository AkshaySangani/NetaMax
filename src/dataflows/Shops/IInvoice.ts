export interface IInvoice {
  /**
   * The Orders's Id
   * @type {string}
   *
   * */
  id: string;

  /**
   * The Url PDF Invoice
   * @type {String}
   *
   * */
  pdfUrl: string;

  /**
   * The Order's delivery date
   * @type {String}
   *
   * */
  invoiceDate: string;

  /**
   * The Order's list
   * @type {String}
   *
   * */
  orders: string;

  /**
   * The Invoice created Date
   * @type {String}
   *
   * */
  createdOnUtc: string;

  /**
   * The StoreId owner of invoice
   * @type {String}
   *
   * */
  storeId: string;

  /**
   * The Delivery Status
   * @type {String}
   *
   * */
  deliveryStatus: string;

  /**
   * The subtotal Invoice
   * @type {number}
   *
   * */
  subTotal: number;

  /**
   * The Invoice Disscount
   * @type {number}
   *
   * */
  //TODO: Check with business the grammar of 'Disscount'
  orderDisccount: number;

  /**
   * The Store Commision
   * @type {number}
   *
   **/
  //TODO: Check with business the grammar of 'Commisssion'
  storeCommisssion: number;
}
