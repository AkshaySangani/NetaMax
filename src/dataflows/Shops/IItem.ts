export interface IItem {
  /**
   * The items's name
   * @type {String}
   *
   * */
  name: string;

  /**
   * The items's quantity bought
   * @type {String}
   *
   * */
  qty: number;

  /**
   * The items's price
   * @type {Number}
   *
   * */
  price: number;

  /**
   * The items's image
   * @type {String}
   *
   * */
  image: string;

  /**
   * The items's image format
   * @type {String}
   *
   * */
  imageFormat: string;
}
