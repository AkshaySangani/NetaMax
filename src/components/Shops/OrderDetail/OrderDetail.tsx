import { ReactElement } from "react";

import { CURRENCY_NAME } from "constants/localeFormatConstants";
import { IItem } from "dataflows/Shops/IItem";
import { formatMoney } from "utils/currencyUtils";

import { Box, Container, Flex, Img, Text } from "@chakra-ui/react";

/**
 * The Order Detail Component
 * @param {IItem} props the component props
 * @returns {ReactElement} The Deliver Order Detail
 */
export const OrderDetail = (props: IItem): ReactElement => {
  const { image, name, qty, price } = props;

  return (
    <Container borderBottom="0.5px solid rgba(53, 53, 53, 0.2)" paddingY="24px">
      <Flex alignItems="center" justify="space-between">
        <Flex w="80px" justify="center">
          <Img maxH="80px" maxW="80px" alignContent="center" src={image} alt={name} fit="contain" />
        </Flex>
        <Box>
          <Text fontSize="16px" fontWeight="800" textAlign="right" maxWidth="188px">
            {name}
          </Text>
          <Text fontSize="14px" fontWeight="400" textAlign="right">
            {qty} {`unidad${qty > 1 ? "es" : ""}`}
          </Text>
          <Text marginTop="26px" fontSize="16px" fontWeight="600" textAlign="right">
            {formatMoney(Number(price), CURRENCY_NAME)}
          </Text>
        </Box>
      </Flex>
    </Container>
  );
};
