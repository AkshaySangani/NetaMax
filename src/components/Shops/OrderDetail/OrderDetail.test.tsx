import { render, cleanup, screen } from "@testing-library/react";
import { OrderDetail } from "./OrderDetail";
import { IItem } from "dataflows/Shops/IItem";

describe("OrderDetail component", () => {
  let mock: IItem;

  beforeEach(() => {
    mock = {
      imageFormat: "image format",
      image: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0003878_0.webp",
      name: "Lampara de UÃ±as Uv-Led Sun Secado Rapido 54w",
      price: 149.0,
      imageFormat: "image format",
      qty: 1,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("OrderDetail should be render correctly", () => {
    const { container } = render(<OrderDetail {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Image should be render properly", () => {
    render(<OrderDetail {...mock} />);
    expect(screen.getByRole("img")).toHaveAttribute("src", mock.image);
  });

  test("Name should be render properly", () => {
    render(<OrderDetail {...mock} />);
    expect(screen.getByText(mock.name)).toBeInTheDocument();
  });
});
