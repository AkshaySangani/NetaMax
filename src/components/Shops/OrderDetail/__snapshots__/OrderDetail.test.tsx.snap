// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail component OrderDetail should be render correctly 1`] = `
<div>
  <div
    class="chakra-container css-1f810hq"
  >
    <div
      class="css-1lekzkb"
    >
      <div
        class="css-x4l7hw"
      >
        <img
          alt="Lampara de Uñas Uv-Led Sun Secado Rapido 54w"
          class="chakra-image css-1g7okjh"
          fit="contain"
          src="https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0003878_0.webp"
        />
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-wbyk11"
        >
          Lampara de Uñas Uv-Led Sun Secado Rapido 54w
        </p>
        <p
          class="chakra-text css-1s07bbm"
        >
          1
           
          unidad
        </p>
        <p
          class="chakra-text css-171oaxm"
        >
          $149.00
        </p>
      </div>
    </div>
  </div>
</div>
`;
